app/controllers/collections_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CollectionsController < AuthenticateController
+  def create
+    collection = Collection.new(collection_params.except(:pocket_serial_numbers).merge(route: route))
+    serials = params[:collection][:pocket_serial_numbers]
+    pockets = serials.collect { |s| { serial_number: s, organization: logged_user.organization } }
+    collection.pockets.new(pockets)
```

Comment:
Research about `nested_attributes`. It is possible to make this easier, if pockets is accepted as a nested attribute for collections, you can create them by putting pockets params inside collection params. If you don't get my point, let me know and we discuss it.